src/adagrams.js

```diff
@@ -1,15 +1,146 @@
+//Get an array of letters, where every letter repeats depending on its frequency
+
+// import { splice } from "core-js/core/array";
+
+// can be done by hand (actually it would take less time)
+export const buildArrayLetters = function (letterDistribution) {
+  const lettersList = [];
+  for (const letter in letterDistribution) {
+    while (letterDistribution[letter] > 0) {
+      lettersList.push(letter);
+      letterDistribution[letter] = letterDistribution[letter] - 1;
+    }
+  }
+  return lettersList;
+};
```

Comment:
This function mutates the input, which may be surprising to the caller. Consider renaming the function so it's clearer that the input is mutated or make a copy of the letter distribution to avoid mutation.